llmfoundry/utils/config_utils.py

```diff
@@ -116,6 +116,25 @@ def process_init_device(model_cfg: DictConfig, fsdp_config: Optional[Dict]):
             # Set defaults for mixed initialization
             fsdp_config.setdefault('use_orig_params', False)
             fsdp_config.setdefault('load_monolith_rank0_only', True)
+
+    # no mixed precision needed for weights when they're already 16 bits
+    master_dtype = model_cfg.get('master_weights_dtype')
+    small_dtypes = ('bf16', 'f16', 'float16', 'bfloat16', 'amp_fp16',
```

Comment:
here and in train.py, you define magic strings (representing dtypes)
can these be thrown into a `llmfoundry/util/precision.py` files (or something like that)

I dont like that composer already has things like this
we can do better.

Reply:
turned this into [a JIRA issue](https://mosaicml.atlassian.net/browse/RESEARCH-804)